tests/test_ecephys/test_tools_spikeinterface.py

```diff
@@ -1009,6 +1028,173 @@ def test_write_units_table_in_processing_module(self):
+            np.testing.assert_array_almost_equal(wf_mean_si, wf_mean_nwb)
+            wf_sd_si = we.get_template(unit_ids[unit_index], mode="std")
+            wf_sd_nwb = nwbfile.units[unit_index]["waveform_sd"].values[0]
+            np.testing.assert_array_almost_equal(wf_sd_si, wf_sd_nwb)
+
+    def test_write_single_segment(self):
+        """This test that the waveforms are written appropriately for the single segment case"""
+        write_waveforms(waveform_extractor=self.single_segment_we, nwbfile=self.nwbfile1, write_electrical_series=True)
+        self._test_waveform_write(self.single_segment_we, self.nwbfile1)
+        self.assertIn("ElectricalSeriesRaw", self.nwbfile1.acquisition)
+
+    def test_write_multiple_segments(self):
+        """This test that the waveforms are written appropriately for the multi segment case"""
+        write_waveforms(waveform_extractor=self.multi_segment_we, nwbfile=self.nwbfile2, write_electrical_series=False)
+        self._test_waveform_write(self.multi_segment_we, self.nwbfile2)
+
+    def test_write_subset_units(self):
+        """This test that the waveforms are sliced properly based on unit_ids"""
+        subset_unit_ids = self.single_segment_we.unit_ids[::2]
+        write_waveforms(waveform_extractor=self.single_segment_we, nwbfile=self.nwbfile3, unit_ids=subset_unit_ids)
+        self._test_waveform_write(self.we_slice, self.nwbfile3, test_properties=False)
+
+        self.assertEqual(len(self.nwbfile3.units), len(subset_unit_ids))
+        self.assertTrue(all(str(unit_id) in self.nwbfile3.units["unit_name"][:] for unit_id in subset_unit_ids))
+
+    def test_write_recordingless(self):
+        """This test that the waveforms are sliced properly based on unit_ids"""
+        write_waveforms(
+            waveform_extractor=self.we_recless,
+            nwbfile=self.nwbfile4,
+            recording=self.we_recless_recording,
+            write_electrical_series=True,
+        )
+        self._test_waveform_write(self.we_recless, self.nwbfile4, test_properties=False)
+
+        # check that not passing the recording raises and Exception
+        with self.assertRaises(Exception) as context:
+            write_waveforms(
+                waveform_extractor=self.we_recless,
+                nwbfile=self.nwbfile4,
+                recording=self.we_recless_recording,
+                write_electrical_series=True,
+            )
+
+    def test_cleanup_tmp_properties(self):
+        # test that if write_waveforms fails, temporary properties are cleaned up
+        try:
+            write_waveforms(waveform_extractor=self.we_recless, nwbfile=self.nwbfile5, recording=None)
+        except:
+            sorting_properties = self.we_recless.sorting.get_property_keys()
+            self.assertNotIn("peak_to_valley", sorting_properties)
+            self.assertNotIn("amplitud_cutoff", sorting_properties)
+
+    def write_test_files(self):
```

Comment:
Are these being tested even if the name is not starting with test? 

Also, I am confused why you create five nwb files in the `setUp` but per test you really only use one file, correct?

Reply:
the `write_*` function was done to save the created files for inspection. For the same reason, I made more files (so that I could save them with the proper names). I'll remove them and keep the usual design